src/algebra/non_unital_alg_hom.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import algebra.algebra.basic
+
+/-!
+# Morphisms of non-unital algebras
+
+This file defines morphisms between two types, each of which carries an addition, additive zero,
+multiplication, and scalar action. The multiplications are not assumed to be associative or
+unital, or even to be compatible with the scalar multiplications. In a typical application, the
+operations will satisfy compatibility conditions making them into algebras (albeit possibly
+non-associative and/or non-unital) but such conditions are not required to make this definition.
```

Comment:
Maybe mention an example application?

Reply:
OK I've just pushed https://github.com/leanprover-community/mathlib/pull/7863/commits/76ec6cc51c28b694afc9016e27edb2b005f72914 which gives on immediate and one future example application.